pkg/handlers/webhookreceiver.go

```diff
@@ -211,6 +211,12 @@ func (h *WebhookReceiverHandler) processAlert(alert template.Alert, mnl *oav1alp
 		return err
 	}
 
+	status, err := managedNotifications.Status.GetNotificationRecord(notification.Name)
+	if err != nil {
+		return err
+	}
+	metrics.SetTotalServiceLogCount(notification.Name, status.ServiceLogSentCount+1)
```

Comment:
I've not tested this at all, so I could be wrong.. but won't `ServiceLogSentCount` will have already been incremented by the earlier `updateNotificationStatus` (which increments it via the `SetNotificationRecord` function)?

Reply:
I tested again, the value set to metric will be 1 less than the one in the managedNotification status.
I think maybe the value fetched in the previous step was not updated one?

But I think there is one problem with the existing format, I made a small change here.

Reply:
As noted in our separate discussion, the problem is that updateNotificationStatus [0] does not update the managed notification it is supplied as a parameter, nor does it return an updated one, so using it for the subsequent GetNotificationRecord will result in an error the very first time an SL is sent for a template

We can fix the way that function works to obtain that behaviour and then don't need to do the +1 correction. Because the other problem with the code as-is without that fix is that the very first time an SL is sent the metric won't be set at all (as GetNotificationRecord will fail). It only starts being set the second time.

[0] https://github.com/openshift/ocm-agent/blob/master/pkg/handlers/webhookreceiver.go#L338